src/Compilers/CSharp/Portable/Binder/Binder_Attributes.cs

```diff
@@ -277,13 +277,19 @@ protected virtual bool IsAttributeConditionallyOmitted(NamedTypeSymbol attribute
                 ArrayBuilder<BoundExpression> boundNamedArgumentsBuilder = null;
                 HashSet<string> boundNamedArgumentsSet = null;
 
+                // VariableUsePass assumes we're not in a scope that can reference local variables.
+                // Need to rework VariableUsePass if that ever changes.
+                Debug.Assert((this.ContainingMemberOrLambda as MethodSymbol)?.MethodKind != MethodKind.LocalFunction);
```

Comment:
> Obviously tests that were added are not adequate, they should be testing behavior of the compiler rather than trying to prove that certain asserts do not fire.

I'm not sure what exactly your feedback is here. You are suggesting I make a test change to "test the behavior of the compiler", but I have no understanding of what you mean by that, or how that is different from "trying to prove that certain asserts do not fire". I would appreciate clarification on what you have in mind that I do to resolve this feedback instead of stating the obviousness inadequacy of the tests.

Additionally, while I understand I could ask you in-person for clarification, that both hides the clarification from other reviewers, and me lacking knowledge of your intent prevents me from looking into the issue beforehand and asking relevant questions (via a background of understanding), as you requested. Please keep that in mind in the future, thanks!

---

On a different note, I suggested marking fields as used during binding of local function default parameters - this would *significantly* simplify the implementation and remove many edge cases (such as this one). However, you noted that an unused local function would mark its referenced fields as used - but we have the same behavior in other places. For example:

    static int field = 2;
    static string unused = nameof(field);

`unused` is marked with `warning CS0414: The field 'Program.unused' is assigned but its value is never used`, but `field` is *not* marked as unused - i.e. we do not do transitive unused analysis. As such, I'm inclined to have the same behavior for default parameters of unused local functions referencing fields. Do you know of a reason why that would be unacceptable behavior?

Another example of the existing behavior:

    static int field = 2; // NOT marked as unused
    static void Main()
    {
        if (false)
        {
            var unused = nameof(field); // marked as unreachable and unused
        }
    }